File: single-language-analysis/dataset-files/python/test_smell_148.py
# Original URL: https://github.com/wookayin/gpustat/blob/7a3b9644229681a752b560c6d4889929b9ea39d5/gpustat/test_gpustat.py#L556-L573

class TestGPUStat(object):

    def test_smell_148(self, scenario_nonexistent_pid):
        """
        Test a case where nvidia query returns non-existent pids (see #16, #18)
        for GPU index 2.
        """
        fp = StringIO()

        gpustats = gpustat.new_query()
        gpustats.print_formatted(fp=fp)

        ret = fp.getvalue()
        print(ret)

        # gpu 2: should ignore process id
        line = remove_ansi_codes(ret).split('\n')[3]
        assert '[2] GeForce RTX 2' in line, str(line)
        assert '99999' not in line
        assert '(Not Supported)' not in line

File: single-language-analysis/dataset-files/java/TestSmell148.java
import org.junit.Test;
import static org.junit.Assert.*;

public class TestSmell148 {

    @Test
    public void testNewQueryMockedNonexistentPid() {
        StringBuilder fp = new StringBuilder();

        GPUSStat gpustats = GPUSStat.newQuery();
        gpustats.printFormatted(fp);

        String ret = fp.toString();
        System.out.println(ret);

        String line = removeAnsiCodes(ret).split("\n")[3];
        assertTrue(line.contains("[2] GeForce RTX 2"), line);
        assertFalse(line.contains("99999"));
        assertFalse(line.contains("(Not Supported)"));
    }
}